lms/static/scripts/frontend_apps/components/EmailNotificationsPreferences.tsx

```diff
@@ -44,63 +77,83 @@ export default function EmailNotificationsPreferences({
   const selectAll = useCallback(() => setAllTo(true), [setAllTo]);
   const selectNone = useCallback(() => setAllTo(false), [setAllTo]);
 
-  return (
-    <Panel
-      onClose={onClose}
-      title="Email Notifications"
-      buttons={<Button variant="primary">Save</Button>}
-    >
-      <p className="font-bold">
-        Receive email notifications when your students annotate.
-      </p>
-      <p className="font-bold">Select the days you{"'"}d like your emails:</p>
+  const onSubmit = useCallback(
+    (event: Event) => onSave({ selectedDays, event }),
+    [onSave, selectedDays]
+  );
 
-      <div className="flex justify-between px-4">
-        <div className="flex flex-col gap-1">
-          {dayNames.map(([day, name]) => (
-            <span
-              key={day}
-              className={classnames(
-                // The checked icon sets fill from the text color
-                'text-grey-6'
-              )}
-            >
-              <Checkbox
-                checked={selectedDays[day]}
-                onChange={() =>
-                  updateSelectedDays({ [day]: !selectedDays[day] })
-                }
-                data-testid={`${day}-checkbox`}
-              >
-                <span
-                  className={classnames(
-                    // Override the color set for the checkbox fill
-                    'text-grey-9'
-                  )}
-                >
-                  {name}
-                </span>
-              </Checkbox>
-            </span>
-          ))}
-        </div>
-        <div className="flex items-start gap-2">
-          <Button
-            variant="secondary"
-            onClick={selectAll}
-            data-testid="select-all-button"
-          >
-            Select all
-          </Button>
+  return (
+    <form onSubmit={onSubmit} method="post" action="">
+      <Panel
+        onClose={onClose}
+        title="Email Notifications"
+        buttons={
           <Button
-            variant="secondary"
-            onClick={selectNone}
-            data-testid="select-none-button"
+            variant="primary"
+            type="submit"
+            disabled={saving}
+            data-testid="save-button"
           >
-            Select none
+            {saving ? 'Saving…' : 'Save'}
```

Comment:
I would suggest to just set the value to "Save" always here, since you've already implemented disabling the button, and the browser UI will indicate that a page load is in progress.